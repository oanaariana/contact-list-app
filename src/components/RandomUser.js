import React, { Component } from 'react';

class RandomUser extends Component {
  render() {
    return (
      <div className="wrapper__box">
        <h4 className="underline" onClick={() => this.props.onAccordionClick(this.props.id)}>{`${this.props.user.name.last}  ${this.props.user.name.first}`}</h4>
        <div key={this.props.id} id={this.props.id}
          className={`check-card ${this.props.isOpen ? "visible" : "unvisible"}`}>
          <div className="container">
            <button className="container__closebtn" onClick={() => this.props.onAccordionClick(this.props.id)}>X</button>
            <img src={this.props.user.picture.medium} className="container__image" alt="" />
            <div className="container__text">
              <h4 className="container__text--title">{`${this.props.user.name.first} ${this.props.user.name.last}`}</h4>
              <table className="container__text--content">
                <tbody>
                  <tr>
                    <td><strong>e-mail</strong></td>
                    <td>{this.props.user.email}</td>
                  </tr>
                  <tr>
                    <td><strong>phone</strong></td>
                    <td>{this.props.user.phone}</td>
                  </tr>
                  <tr>
                    <td><strong>street</strong></td>
                    <td>{`${this.props.user.location.street.name} ${this.props.user.location.street.number}`}</td>
                  </tr>
                  <tr>
                    <td><strong>city</strong></td>
                    <td>{this.props.user.location.city}</td>
                  </tr>
                  <tr>
                    <td><strong>state</strong></td>
                    <td>{this.props.user.location.state}</td>
                  </tr>
                  <tr>
                    <td className="mark"><strong>postcode</strong></td>
                    <td>{this.props.user.location.postcode}</td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default RandomUser;