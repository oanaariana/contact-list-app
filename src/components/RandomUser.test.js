import React from "react";
import RandomUser from "./RandomUser";

import { render } from "@testing-library/react";

const stubbedUser = {
  gender: "female",
  name: {
    "title": "Miss",
    "first": "Bella",
    "last": "Wang"
  },
  location: {
    street: {
      number: 9617,
      name: "Stafford Street"
    },
    city: "Hastings",
    state: "Wellington",
    country: "New Zealand",
    postcode: 66909,
    coordinates: {
      latitude: "17.2334",
      longitude: "88.9196"
    },
    timezone: {
      offset: "+4:30",
      description: "Kabul"
    }
  },
  email: "bella.wang@example.com",
  login: {
    uuid: "2d5afd5e-a598-469f-9933-4edba18bcefb",
    username: "blueelephant921",
    password: "ccccccc",
    salt: "uJHyJ9ZL",
    md5: "5724c4fbd97e5155d19f71a44f12cc48",
    sha1: "85dff52890cee9977409fc6ae332a7e2326d6aa4",
    sha256: "ab62d511f943bfca48935cb4620b3b2007f8d1b9c1e04a04ea76ea95cd099d5f"
  },
  dob: {
    date: "1989-11-17T21:56:57.779Z",
    age: 32
  },
  registered: {
    date: "2014-05-20T20:54:39.692Z",
    age: 7
  },
  phone: "(057)-889-6313",
  cell: "(645)-978-7056",
  id: {
    name: "",
    value: null
  },
  picture: {
    large: "https://randomuser.me/api/portraits/women/62.jpg",
    medium: "https://randomuser.me/api/portraits/med/women/62.jpg",
    thumbnail: "https://randomuser.me/api/portraits/thumb/women/62.jpg"
  },
  nat: "NZ"
};

it("matches snapshot", () => {
  const { asFragment } = render(<RandomUser user={stubbedUser} />);

  expect(asFragment()).toMatchSnapshot();
});
