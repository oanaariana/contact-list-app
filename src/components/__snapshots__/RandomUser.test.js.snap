// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper__box"
  >
    <h4
      class="underline"
    >
      Bella Wang
    </h4>
    <div
      class="edit-card unvisible"
    >
      <div
        class="container"
      >
        <button
          class="container__closebtn"
        >
          X
        </button>
        <img
          alt=""
          class="container__image"
          src="https://randomuser.me/api/portraits/med/women/62.jpg"
        />
        <div
          class="container__text"
        >
          <h4
            class="container__text--title"
          >
            Bella Wang
          </h4>
          <table
            class="container__text--content"
          >
            <tbody>
              <tr>
                <td>
                  <strong>
                    e-mail
                  </strong>
                </td>
                <td>
                  bella.wang@example.com
                </td>
              </tr>
              <tr>
                <td>
                  <strong>
                    phone
                  </strong>
                </td>
                <td>
                  (057)-889-6313
                </td>
              </tr>
              <tr>
                <td>
                  <strong>
                    street
                  </strong>
                </td>
                <td>
                  Stafford Street 9617
                </td>
              </tr>
              <tr>
                <td>
                  <strong>
                    city
                  </strong>
                </td>
                <td>
                  Hastings
                </td>
              </tr>
              <tr>
                <td>
                  <strong>
                    state
                  </strong>
                </td>
                <td>
                  Wellington
                </td>
              </tr>
              <tr>
                <td
                  class="mark"
                >
                  <strong>
                    postcode
                  </strong>
                </td>
                <td>
                  66909
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
